File: src/app/models/clientquiz.ts
import { Color } from './color';
import { Answer } from './answer';
import { Question } from './question';
import { Session } from './session';
import { environment } from '../../environments/environment.prod';

export class ClientQuiz {
    constructor() {
        this.id = environment.emptyGuid;
        this.questionId = environment.emptyGuid;
        this.sessionId = environment.emptyGuid;
    }
    id: string;
    session: Session;
    sessionId: string;
    order: number;
    selectedAnswer: Answer;
    selectedAnswerId: string;
    question: Question;
    questionId: string;
    creationDate: Date;
    deletionDate: Date;
    isDeleted: boolean;
}

File: API/Models/ClientQuiz.cs
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System;
using System.ComponentModel.DataAnnotations;

namespace QuizAppApi.Models
{
    public class ClientQuiz : BaseEntity
    {
        [ForeignKey("SessionId")]
        public Session Session { get; set; }
        public Guid SessionId { get; set; }
        public int Order { get; set; }
        [ForeignKey("SelectedAnswerId")]
        public Answer SelectedAnswer { get; set; }
        public Guid? SelectedAnswerId { get; set; }
        [ForeignKey("QuestionId")]
        public Question Question { get; set; }
        public Guid QuestionId { get; set; }
    }
}
